integration-cli/docker_cli_swarm_test.go

```diff
@@ -1446,6 +1446,12 @@ func (s *DockerSwarmSuite) TestSwarmClusterRotateUnlockKey(c *check.C) {
 			c.Assert(getNodeStatus(c, d), checker.Equals, swarm.LocalNodeStateActive)
 
 			outs, err = d.Cmd("node", "ls")
+			if err != nil {
```

Comment:
What's the purpose of those new added code block?

Reply:
The test  seems to assume something not clearly defined, in this case, how soon should the daemon node rejoin as a manager after restart.   I just add some delay and then make a retry. In previous code change, I use 500ms delay and the test still report FAIL for 1 jobs. Current code change use a 10s delay and it seems PASS all test jobs at least for one time. And I clean up those commits and push new ones, now new round of test is started. Finger crossed.

Reply:
We might prefer to have this change/discussion in a different PR, wdyt @thaJeztah ?

Reply:
Yes, having it in a separate PR would be preferable (so that the PR is focused on just the changes needed for the race condition)